tutorial/Lab2/README.md

```diff
@@ -0,0 +1,151 @@
+    pod/guestbook-v1-7fc76dc46-hspnk    1/1       Running   0          1h
+    pod/guestbook-v1-7fc76dc46-sxzkt    1/1       Running   0          1h
+    pod/redis-master-5d8b66464f-pvbl9   1/1       Running   0          1h
+    
+    NAME                   TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)          AGE
+    service/guestbook      NodePort    172.21.45.29    <none>        3000:31989/TCP   31s
+    service/kubernetes     ClusterIP   172.21.0.1      <none>        443/TCP          9d
+    service/redis-master   ClusterIP   172.21.232.61   <none>        6379/TCP         1h
+    
+    NAME                           DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+    deployment.apps/guestbook-v1   3         3         3            3           1h
+    deployment.apps/redis-master   1         1         1            1           1h
+    
+    NAME                                      DESIRED   CURRENT   READY     AGE
+    replicaset.apps/guestbook-v1-7fc76dc46    3         3         3         1h
+    replicaset.apps/redis-master-5d8b66464f   1         1         1         1h
+    ```
+    Note: The service type has changed (to `NodePort`) and a new port has been allocated (`31989` in this output case) to the guestbook 
+    service. All `redis-slave` resources have been removed.
+
+5. View the guestbook as per [Lab1](../Lab1/README.md), using the updated port for the guestbook service
+   
+# Update the application using Helm
+
+In this part of the lab we will update the previously deployed application `guestbook-demo`, using Helm.
+
+Before we start, let's take a few minutes to see how does Helm simplify the process compared to using Kubernetes directly. Helms' use of a [template language](https://docs.helm.sh/chart_template_guide/) provides great flexibility and power to chart authors which removes the complexity to the chart user. In the guestbook example, we are using the following capabilities of templating:
+* Values: which is an object that provides access to values passed into the chart. An example of this is in `guestbook-service` which contains the following line: ```type: {{ .Values.service.type }}```. The line provides the capability to set the service type during upgrade or install.
+* Control structures: (called “actions” in template parlance) provides the template author with the ability to control the flow of a template’s generation. An example of this is in `redis-slave-service` which contains the following line: ```{{- if .Values.redis.slaveEnabled -}}```. The line provides the capability to enable/disable the REDIS master/slave during upgrade or install.
+
+Theory over, let's give it a go!
+
+1. Update the application
+
+    ```helm upgrade guestbook-demo ./guestbook --set redis.slaveEnabled=false,service.type=NodePort --namespace helm-demo```
+    
+    Helm upgrade takes an existing release and upgrades it according to the information you provide. You should see output similar to the following:
```

Comment:
This looks like it's still code? Formatting?

Reply:
```A Helm upgrade takes an existing release and upgrades it according to the information you provide.```

Reply:
done